### PR TITLE
Update documentation round 3

### PR DESCRIPTION
**Pull request by mjlbach**
I merged the majority of the changes, which I think were already positive. Copying the feedback here.

levouh: You think there should be a table of contents, but possibly github's auto TOC mitigates this

levouh: You think we should add a little bit of wording above each mapped function to explain what they do (already did some of this)

David-Else: You think "Automatically launching language servers" should be renamed to explicitly call out root dir

### CODE REVIEW

**Comment by David-Else** [comment]
> David-Else: You think "Automatically launching language servers" should be renamed to explicitly call out root dir

You gave a reason to not do that, so it is up to you. I was reading it from a technical writing point of view rather than with the knowledge you have about issues.

**Comment by levouh** [comment]
Pasting from my smooth-brain reply in the old PR.

---

>You think there should be a table of contents, but possibly github's auto TOC mitigates this

@mjlbach the built-i one from GitHub is nice, but I'm not sure how many people are aware it exists. You can add it manually, or do something like what @David-Else noted with `pandoc` or similar. I can't imagine it makes a _huge_ difference, but like I said it just helps cater to interest level more easily.

>You think we should add a little bit of wording above each mapped function to explain what they do

Sort of, my only comment being that `nvim-lspconfig` is often people's first introduction to the LSP client _at all_, so noting some of the cool features (in plain English) as well as making it apparent where to learn more (like you said, `:help vim.lsp.*`) might help shrink that gap in understanding.

**Comment by levouh** [comment]
There are also a few things like the random `Window` tidbits at the bottom of the _README_ and whatnot that seem a bit out of place.

**Comment by mjlbach** [comment]
> There are also a few things like the random Window tidbits at the bottom of the README and whatnot that seem a bit out of place.

Not sure what to do about this, I get a lot of questions about windows support though. I've been resisting just doing a tree-wide if windows then cmd = cmd .. ".cmd" end type thing
